Build log not printed when error_code is -1

When the build errors out with a status code of -1, the output isn't printed

Example output of `vcpkg install`:

```
-- Configuring x64-windows-dbg
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:59 (message):
  Failed to execute command "C:/Program Files/CMake/bin/cmake.exe --build .
  --config Debug -- -v -j33" in working directory
  "C:/Users/xxx/git/vcpkg/buildtrees/my-package/x64-windows-dbg":
  -1
```

I tracked it down to this bit of code:

```cmake
    execute_process(
        COMMAND ${arg_COMMAND}
        WORKING_DIRECTORY "${arg_WORKING_DIRECTORY}"
        OUTPUT_FILE "${log_out}"
        ERROR_FILE "${log_err}"
        RESULT_VARIABLE error_code
    )
    if (NOT error_code MATCHES "^[0-9]+$")
        list(JOIN arg_COMMAND " " command)
        message(FATAL_ERROR "Failed to execute command \"${command}\" in working directory \"${arg_WORKING_DIRECTORY}\": ${error_code}")
    endif()
    if(NOT error_code EQUAL "0")
        file(READ "${log_out}" out_contents)
        file(READ "${log_err}" err_contents)
        set(all_contents "${out_contents}${err_contents}")
        if(all_contents MATCHES "${Z_VCPKG_EXECUTE_BUILD_PROCESS_RETRY_ERROR_MESSAGES}")
   ...
```

I think the condition `if (NOT error_code MATCHES "^[0-9]+$")` should allow `-1` strings to go to the next condition

--- a/scripts/cmake/vcpkg_execute_build_process.cmake
+++ b/scripts/cmake/vcpkg_execute_build_process.cmake
@@ -53,7 +53,7 @@ function(vcpkg_execute_build_process)
         ERROR_FILE "${log_err}"
         RESULT_VARIABLE error_code
     )
-    if (NOT error_code MATCHES "^[0-9]+$")
+    if (NOT error_code MATCHES "^-?[0-9]+$")
         list(JOIN arg_COMMAND " " command)
         message(FATAL_ERROR "Failed to execute command \"${command}\" in working directory \"${arg_WORKING_DIRECTORY}\": ${error_code}")
     endif()

